app/src/test/java/io/appium/uiautomator2/model/settings/AbstractSettingTests.java

```diff
@@ -20,57 +20,99 @@
 
 import io.appium.uiautomator2.common.exceptions.UiAutomator2Exception;
 
+import static org.mockito.ArgumentMatchers.anyInt;
 import static org.mockito.ArgumentMatchers.anyLong;
 import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.verify;
 import static org.mockito.MockitoAnnotations.initMocks;
 
 public class AbstractSettingTests {
-
-    @Rule
-    public ExpectedException thrown = ExpectedException.none();
+    @Spy
+    private DummyIntegerSetting dummyIntegerSetting;
 
     @Spy
-    private DummySetting dummySetting;
+    private DummyLongSetting dummyLongSetting;
 
     @Before
     public void setUp() {
         initMocks(this);
     }
 
     @Test
-    public void shouldThrowExceptionIfTypeIsNotValid() {
-        thrown.expect(UiAutomator2Exception.class);
-        thrown.expectMessage("Invalid setting value type. Got: java.lang.String. Expected: java.lang.Long");
-        dummySetting.update("test");
+    public void dummyIntegerSettingShouldThrowExceptionIfTypeIsNotValid() {
+        try {
+            dummyIntegerSetting.update("test");
+        } catch (UiAutomator2Exception e) {
+            Assert.assertEquals("Invalid setting value type. Got: java.lang.String. Expected: java.lang.Integer", e.getMessage());
+        }
     }
 
     @Test
-    public void shouldNotThrowExceptionIfApplyFailed() {
-        doThrow(new UiAutomator2Exception("error")).when(dummySetting).apply(anyLong());
-        dummySetting.update(10);
+    public void dummyIntegerSettingShouldNotThrowExceptionIfApplyFailed() {
+        doThrow(new UiAutomator2Exception("error")).when(dummyIntegerSetting).apply(anyInt());
+        dummyIntegerSetting.update(10);
+    }
+
+    @Test
+    public void dummyIntegerSettingShouldReturnValidValueType() {
+        Assert.assertEquals(Integer.class, dummyIntegerSetting.getValueType());
+    }
+
+    @Test
+    public void dummyIntegerSettingShouldCallApplyWithValidValue() {
+        dummyIntegerSetting.update(123);
+        verify(dummyIntegerSetting).apply(123);
+    }
+
+    @Test
+    public void dummyLongSettingShouldThrowExceptionIfTypeIsNotValid() {
+        try {
+            dummyLongSetting.update("test");
+        } catch (UiAutomator2Exception e) {
+            Assert.assertEquals("Invalid setting value type. Got: java.lang.String. Expected: java.lang.Long", e.getMessage());
+        }
+    }
+
+    private class DummyIntegerSetting extends AbstractSetting<Integer> {
+        public DummyIntegerSetting() {
+            super(Integer.class, "dummyInteger");
+        }
+
+        @Override
+        public Integer getValue() {
+            return 123;
+        }
+
+        @Override
+        protected void apply(Integer value) {
+            // do nothing
+        }
     }
 
     @Test
-    public void shouldReturnValidValueType() {
-        Assert.assertEquals(Long.class, dummySetting.getValueType());
+    public void dummyLongSettingShouldNotThrowExceptionIfApplyFailed() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dpgraham/appium-uiautomator2-server/pullRequest?prid=5553523)